### PR TITLE
feat(coderd): add endpoints for editing and deleting organizations

### PR DESCRIPTION
**Pull request by aslilac**
Add backend and codersdk support for editing and deleting organizations

- `PATCH /api/v2/organizations/:orgId` for updating an existing org
- `DELETE /api/v2/organizations/:orgId` for deleting an org
- `codersdk.Client.UpdateOrganization` to match the new `PATCH` endpoint
- `codersdk.Client.DeleteOrganization` to match the new `DELETE` endpoint
- also `codersdk.Client.CreateOrganization`, because it was missing

### CODE REVIEW

File: .vscode/settings.json
Comment (aslilac) [inline .vscode/settings.json:198]: this settings affects the quick open menu, which I use heavily to navigate the codebase, and which means I have to go to great lengths just to open typesGenerated.ts. I quite dislike these opinionated sorts of configs being baked into the codebase. 😅

File: coderd/database/dbmem/dbmem.go
Comment (Emyrk) [inline coderd/database/dbmem/dbmem.go:7162]:
```suggestion
	err := validateDatabaseType(arg)
	if err != nil {
		return database.Organization{}, err
	}

	q.mutex.Lock()
	defer q.mutex.Unlock()

	for i, org := range q.organizations {
```

File: coderd/database/dbmem/dbmem.go
Comment (Emyrk) [inline coderd/database/dbmem/dbmem.go:1584]:
```suggestion
func (q *FakeQuerier) DeleteOrganization(_ context.Context, id uuid.UUID) error {
	q.mutex.Lock()
	defer q.mutex.Unlock()

	for i, org := range q.organizations {
		if org.ID == id && !org.IsDefault {
			q.organizations = append(q.organizations[:i], q.organizations[i+1:]...)
			return nil
		}
	}
	return sql.ErrNoRows
}
```

File: coderd/database/queries/organizations.sql
Comment (Emyrk) [inline coderd/database/queries/organizations.sql:65]: I prefer names, but your call
```suggestion
-- name: UpdateOrganization :one
UPDATE
	organizations
SET
	updated_at = @updated_at,
	name = @name
WHERE
	id = @id
RETURNING *;
```

File: coderd/organizations.go
Comment (Emyrk) [inline coderd/organizations.go:175]: I know this is annoying, but this catches rbac errors and raises them correctly.

I also added the duplicate org error.
```suggestion
	organization, err := api.Database.UpdateOrganization(ctx, database.UpdateOrganizationParams{
		ID:        organization.ID,
		UpdatedAt: dbtime.Now(),
		Name:      req.Name,
	})
	if httpapi.Is404Error(err) {
		httpapi.ResourceNotFound(rw)
		return
	}
	if database.IsUniqueViolation(err) {
		httpapi.Write(ctx, rw, http.StatusConflict, codersdk.Response{
			Message: "Organization already exists with that name.",
			Validations: []codersdk.ValidationError{{
				Field:  "name",
				Detail: "This value is already in use and should be unique.",
			}},
		})
		return
	}
	if err != nil {
		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
			Message: "Internal error updating organization.",
			Detail:  fmt.Sprintf("update organization: %s", err.Error()),
		})
		return
	}
```

File: coderd/organizations.go
Comment (Emyrk) [inline coderd/organizations.go:161]: Remove this, since you might not be able to fetch the organization anyway.

File: coderd/organizations_test.go
Comment (Emyrk) [inline coderd/organizations_test.go:152]: You can do this for the other tests too
```suggestion
     ctx := testutil.Context(t, testutil.WaitLong)
```

File: scripts/rbacgen/main.go
Comment (Emyrk) [inline scripts/rbacgen/main.go:153]: Rebase to main, I think I fixed these already

**Review by Emyrk — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (aslilac) [reply .vscode/settings.json:198]: the opposite; it was being ignored before, and now will actually show up.

File: coderd/organizations.go
Comment (aslilac) [reply coderd/organizations.go:161]: Not being able to fetch the org is the happy path. This is to check if the new name is already in use, and is taken basically directly from the `postOrganization` handler. I imagine we could remove this and the table constraints would just cause the update to error, but I think I prefer this for the better error messages that it'll give the frontend.

Comment (aslilac) [reply coderd/organizations.go:161]: oh I see, you moved this handling in your other suggestion

**Comment by aslilac** [comment]
So, it turns out there isn't actually a unique constraint on organization names, so I'm adding that in #13311, and will rebase this once that gets merged.

**Review by Emyrk — APPROVED** [review]

**Comment by Emyrk** [comment]
closes https://github.com/coder/coder/issues/12310

**Pull request merged**
